posts/inside-rust/2024-05-07-this-development-cycle-in-cargo-1.79.md

```diff
@@ -0,0 +1,458 @@
+To better communicate this, we borrowed a table from a prior discussion on inheriting default-features
+
+| Workspace | Member | 1.64 behaviour | 1.69 behavior | proposed 202X Edition behavior | Reasoning
+|:---:|:---:|---|---|---|---
+| *nothing* | df=false | Enabled | Enabled, warning that it is ignored | **Disabled** | Basically means let the package control default features when not specified 
+| df=true | df=false  | Enabled | Enabled, warning | Enabled, warning | Should have been an error in the past since the field is ignored. May change to an error in the future.
+| *nothing* | df=true  | Enabled |  Enabled | Enabled | There is no conflict about default being enabled.
+| df=true | df=true | Enabled | Enabled | Enabled | ^Same
+| *nothing* | *nothing*  | Enabled  | Enabled | Enabled | No ambiguity.
+| df=true | *nothing*   | Enabled | Enabled | Enabled | Enabled | ^Same
+| df=false | df=false  | Disabled | Disabled | Disabled | No ambiguity.
+| df=false | df=true  | Disabled | **Enabled** | Enabled | This allows members to decide whether or not they want default features. The workaround of `features=["default"]` seems unnecessary.
+| df=false | df=nothing  | Disabled | Disabled | Disabled | Natural way to write "give me whatever was written in workspace".  ||
+
+Some points that were discussed
+- Documenting this as this deviates from the intended mental model
+- Ecosystem churn in switching to this
+- If we discourage features in workspace dependencies, it might preclude people who are doing it as low effort [workspace hack](https://crates.io/crates/cargo-hakari).  We felt that we should exercise caution and not remove features, yet.
+
+As this was after the cut off for the 2024 Edition,
+this will likely be stuck waiting on the 2027 Edition.
+
+## Misc
+
+- Thank to [PaulDance](https://github.com/PaulDance), we are now publishing `cargo-test-support` and `cargo-test-macro` which can be useful for developing tools that integrate with cargo [#13418](https://github.com/rust-lang/cargo/pull/13418).  The focus is on Cargo's needs, making the APIs and documentation a little rough for others to adopt.
+- Thanks for [linyihai](https://github.com/linyihai), we now have unstable support for `cargo update foo --precise <prerelease>` ([#13626](https://github.com/rust-lang/cargo/pull/13626)), last discussed in [1.76](https://blog.rust-lang.org/inside-rust/2024/01/03/this-development-cycle-in-cargo-1-76.html#rfc-3493-cargo-update---precise-prerelease).
+
+## Focus areas without progress
+
+These are areas of interest for Cargo team members with no reportable progress for this development-cycle.
+
+Ready-to-develop:
+- [Merge `cargo upgrade` into `cargo update`](https://github.com/rust-lang/cargo/issues/12425)
+- [`cargo publish` for workspaces](https://github.com/rust-lang/cargo/issues/1169)
+- [Auto-generate completions](https://github.com/rust-lang/cargo/issues/6645)
+  - See [clap-rs/clap#3166](https://github.com/clap-rs/clap/issues/3166)
```

Comment:
Isn't this already picked up by a GSoC participant? In this context it looks available to claim.

Reply:
Commented out this item